plutus-list/src/PlutusTx/List/Ord.hs

```diff
@@ -0,0 +1,177 @@
+  ordNub,
+  ordNubBy,
+) where
+
+import Data.Function (on)
+import Data.Kind (Type)
+import PlutusTx.Prelude
+import Prelude qualified ()
+
+{- | Checks if the 'Foldable' is sorted with respect to a total order
+represented by a binary relation.
+
+@since 1.0
+-}
+{-# INLINEABLE isSortedBy #-}
+isSortedBy ::
+  forall (f :: Type -> Type) (a :: Type).
+  Foldable f =>
+  (a -> a -> Bool) ->
+  f a ->
+  Bool
+isSortedBy f = snd . foldr go (Nothing, True)
+  where
+    go x (Nothing, _) = (Just x, True)
+    go x (Just y, b) = (Just x, b /\ f x y)
+
+{- | Checks if the 'Foldable' is sorted in ascending order with respect to the
+result of a key function.
+
+@since 1.0
+-}
+{-# INLINEABLE isSortedOn #-}
+isSortedOn ::
+  forall (f :: Type -> Type) (a :: Type) (b :: Type).
+  (Foldable f, Ord b) =>
+  (a -> b) ->
+  f a ->
+  Bool
+isSortedOn f = isSortedBy (\x y -> f x <= f y)
+
+{-# INLINEABLE isSorted #-}
+
+{- | Checks if the 'Foldable' is sorted in ascending order with respect to its
+elements' 'Ord' instance.
+
+@since 1.0
+-}
+isSorted ::
+  forall (f :: Type -> Type) (a :: Type).
+  (Foldable f, Ord a) =>
+  f a ->
+  Bool
+isSorted = isSortedBy (<=)
+
+{- | Checks if the 'Foldable' is sorted strictly in ascending order with respect
+to the result of a key function.. This means any pair of adjacent elements
+cannot be equal.
+
+@since 1.0
+-}
+{-# INLINEABLE isSortedAscendingOn #-}
+isSortedAscendingOn ::
+  forall (f :: Type -> Type) (a :: Type) (b :: Type).
+  (Foldable f, Ord b) =>
+  (a -> b) ->
+  f a ->
+  Bool
+isSortedAscendingOn f = isSortedBy (\x y -> f x < f y)
+
+{- | Checks if the 'Foldable' is sorted strictly in ascending order with respect
+to its elements' 'Ord' instance. This means any pair of adjacent elements cannot
+be equal.
+
+@since 1.0
+-}
+{-# INLINEABLE isSortedAscending #-}
+isSortedAscending ::
+  forall (f :: Type -> Type) (a :: Type).
+  (Foldable f, Ord a) =>
+  f a ->
+  Bool
+isSortedAscending = isSortedBy (<)
+
+{- | The inlineable version of @sort@ in @base@.
+
+ @since 1.0
+-}
+{-# INLINEABLE sort #-}
+sort :: forall (a :: Type). (Ord a) => [a] -> [a]
+sort = sortBy compare
+
+{- | The inlineable version of @sortOn@ in @base@.
+
+ @since 1.0
+-}
+{-# INLINEABLE sortOn #-}
+sortOn :: forall (a :: Type) (b :: Type). (Ord b) => (a -> b) -> [a] -> [a]
+sortOn = sortBy . (compare `on`)
+
+{- | The inlineable version of @sortBy@ in @base@.
+
+ @since 1.0
+-}
+{-# INLINEABLE sortBy #-}
+sortBy :: forall (a :: Type). (a -> a -> Ordering) -> [a] -> [a]
+sortBy cmp = mergeAll . sequences
+  where
+    sequences (a : b : xs)
+      | a `cmp` b == GT = descending b [a] xs
+      | otherwise = ascending b (a :) xs
+    sequences xs = [xs]
+
+    descending a as (b : bs)
+      | a `cmp` b == GT = descending b (a : as) bs
+    descending a as bs = (a : as) : sequences bs
+
+    ascending a as (b : bs)
+      | a `cmp` b /= GT = ascending b (\ys -> as (a : ys)) bs
+    ascending a as bs =
+      let !x = as [a]
```

Comment:
Not sure why `fourmolu` complains about not finding `BangPatterns` even if it is enabled in the cabal file.